60 add organizer

soniahidri opened the pull request:
It is about adding registered user to organizer Collection to be organizer.
to test it : log in with Admin
go to Add Organizer in the menu 
put user email 
to check it go to baasbox / Collection organizer
@trungluongquang I integrated  function to check if the putting email is already registered or not, it's also part of my user story and it make no sense to remove it.

soniahidri committed:
baasbox.js : added function to get user by his email.

soniahidri committed:
services.js : added function to get user by calling the

soniahidri committed:
added new template for add organizer

soniahidri committed:
controllers.js : added AddOrgCtrl

soniahidri committed:
add-organizer.html : added licence to tenmplate.

soniahidri committed:
app.js : added state with parameters for add-organizer

soniahidri committed:
menu.html : added new item for Add Organizer to App menu.

soniahidri committed:
menu.html : added ng-show to Add Organizer item.

soniahidri committed:
controllers.js : Removed unnecessary debug output.

soniahidri committed:
services.js : added new function for adding organizer.

soniahidri committed:
controllers.js : updated AddOrgCtrl, check if the user is already regestred by calling getUsers() then call createOrganizer().

soniahidri committed:
de.json : added translation words

soniahidri committed:
add-organizer.html : updated called function.

soniahidri committed:
commented new functions in services.js, controllers.js and baasbox.js

soniahidri committed:
reformated the code.

soniahidri committed:
services.js : updated getUsers() to be called by registered user, because it is not allowed to change User Role.

soniahidri committed:
controllers.js : deleted state.go

soniahidri committed:
controllers.js : Removed unused Params

trungluongquang committed:
update to the newest master

trungluongquang committed:
update translation

trungluongquang committed:
formatting the code

trungluongquang committed:
format

muradIsa commented:
@soniahidri , implementation of createOrganizer funtion on services.js is not correct. After creating new document in 'organizer' collection, granting all permissions to not existing object in the 'events' collection causes error. Change parameter to 'organizer'. And it would be better also to grant only read permission instead of all permissions, because having all permissions users would be able to assign themselves an organizer role.

trungluongquang committed:
bugfix - collection events to organizer

- keep ALL_PERSMISSION for "registered role" because it is not so
important to change by now. It does not affect other functions and
granting all permission simplifies the deletion of an organizer
function (in case added later)

- change ALL_PERMISSION for "default" user to only READ_PERMISSION to
make sure that there will be no case, a not-logged-in user can remove
the role of an organizer

trungluongquang commented:
as @muradIsa mentioned above, there are some bugs in your pull request. Because we are running out of time so I have fixed them myself. Please check your code carefully before making any pull request and don't forget adding translation (in this pull request, you did forget and I have to add translation)
